core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1651,14 +1665,25 @@ private static void shutdownAndAwaitTermination(ExecutorService pool, String poo
 
     private void removeRemoteTopicPartitionMetrics(TopicIdPartition topicIdPartition) {
         BrokerTopicMetrics topicMetrics = brokerTopicStats.topicStats(topicIdPartition.topic());
+        String topic = topicIdPartition.topic();
         int partition = topicIdPartition.partition();
+        // remove the partition metric values
         topicMetrics.removeRemoteCopyLagBytes(partition);
         topicMetrics.removeRemoteCopyLagSegments(partition);
         topicMetrics.removeRemoteDeleteLagBytes(partition);
         topicMetrics.removeRemoteDeleteLagSegments(partition);
         topicMetrics.removeRemoteLogMetadataCount(partition);
         topicMetrics.removeRemoteLogSizeComputationTime(partition);
         topicMetrics.removeRemoteLogSizeBytes(partition);
+
+        // update the all topic metric values
```

Comment:
suggestion 

change comment to:
Update the all topic metric values so that we have a sample right for all topics metric after removal of partition. In absence of this update, the all topic metric will not update until the next sample point is generated for it.